enos/enos.py

```diff
@@ -53,23 +58,60 @@
 
 
 @enostask("""
-usage: enos up [-f CONFIG_PATH] [--force-deploy] [-t TAGS | --tags=TAGS]
-               [--provider=PROVIDER] [-vv|-s|--silent]
+usage: enos up  [-e ENV|--env=ENV][-f CONFIG_PATH] [--force-deploy]
+                [--provider=PROVIDER] [-t TAGS|--tags=TAGS]
+                [-v|-vv|-s|--silent]
 
 Get resources and install the docker registry.
 
 Options:
-  -h --help            Show this help message.
+  -e ENV --env=ENV     Path to the environment directory. You should
+                       use this option when you want to link to a specific
+                       experiment. Do not specify it in other cases.
   -f CONFIG_PATH       Path to the configuration file describing the
                        deployment [default: ./reservation.yaml].
+  -h --help            Show this help message.
   --force-deploy       Force deployment [default: False].
-  -t TAGS --tags=TAGS  Only run ansible tasks tagged with these values.
   --provider=PROVIDER  The provider name [default: G5k].
+  -t TAGS --tags=TAGS  Only run ansible tasks tagged with these values.
+  -v --version         Show version number.
+  -vv                  Verbose mode.
+  -s --silent          Quiet mode.
 
 """)
 def up(provider=None, env=None, **kwargs):
     logging.debug('phase[up]: args=%s' % kwargs)
 
+    # Generates a directory for results
+    resultdir_name = kwargs['--env'] or \
+            'enos_' + datetime.today().isoformat()
+
+    resultdir = os.path.join(CALL_PATH, resultdir_name)
+    # The result directory cannot be created if a related file exists
+    if os.path.isfile(resultdir):
+        logging.error("Result directory cannot be created due to %s" %
+                resultdir)
+        sys.exit(1)
+
+    # Create the result directory if it does not exist
+    os.path.isdir(resultdir) or os.mkdir(resultdir)
+    logging.info('Generate results in %s' % resultdir)
+    env['resultdir'] = resultdir
+
+    # Symlink the result directory with the current directory
+    link = os.path.abspath(SYMLINK_NAME)
+
+    os.path.lexists(link) and os.remove(link)
+    try:
+        os.symlink(env['resultdir'], link)
+        logging.info("Symlinked %s to %s" % (env['resultdir'], link))
+    except OSError:
+        # An harmless error can occur due to a race condition when multiple
+        # regions are simultaneously deployed
+        logging.warning("Symlink %s to %s failed" %
+                (env['resultdir'], link))
+        pass
+
```

Comment:
I have just modified this part of the code (from `105` to `113`) since I observed that a race condition can occur when multiple regions are deployed simultaneously. More specifically, if `region1` creates the symlink after `region2`reach `104`, `106` would return an error since the link already exists. As pointed out by @rcherrueau, this error is harmless since the `current` directory is not required in the specific context of multiregion deployments.